### DISCUSSION TITLE
Week 9 Reprot

### DISCUSSION THREAD

**Discussion by derrrobin** [SE-TINF22B6/CookPit#48] [General]
In this Sprint we start the connection to our API. Now you can talk to a Ki and tell her what ingredients you have lying around at home. The Ki will then give you a recipe consisting of the ingredients you have named to the Ki. In the future, you can consider saving the recipes you have given if you liked the Ki's recipe so much.

### Explanation of the Function:

The input field is used to specify the ingredients that are to be sent to OpenAI for processing during the subsequent API call. So far, all ingredients are stored in the form of a single string (I am already in the process of changing this). The button is the trigger for the API call and the processing of the ingredients into a recipe.

![image](https://github.com/SE-TINF22B6/CookPit/assets/123726577/7d26fa99-d741-447a-bed1-631550167ae2)

Here is the code for the button, if you are interested.

The function is called by clicking the button.

**Code explanation:**
- The content of the input field is retrieved. If this is empty, no API call is made. If something has been entered, the backend is called and receives the user input. The frontend then waits for a response from the backend. As soon as the frontend receives the response, it is given to the div "return" as textContent so that the response can also be seen on the website.

![image](https://github.com/SE-TINF22B6/CookPit/assets/123726577/7a108df3-0544-4560-b7c3-36cc958d39cb)

If a request is made against the `/recipe-maker` endpoint, the `openai-test` module is called and executed.

![image](https://github.com/SE-TINF22B6/CookPit/assets/123726577/071eb474-2fa2-40ca-8c7c-cc9be714c5be)

In the `openai-test` module, a prefabricated message is built from the user input and a string. This is then sent to the OpenAI API and processed there. The result that is sent back is then sent back to frontend, which, as mentioned above, is finally displayed on the website.

![image](https://github.com/SE-TINF22B6/CookPit/assets/123726577/5e4a11aa-8920-478c-bc46-2113ea405339)

Next, we have continued to work on our start page. At the moment there are no real recipes displayed, but this will be changed in the future. At the moment, there are only pictures. However, we want to keep the design and fill the template with real data as we go along.

![image](https://github.com/SE-TINF22B6/CookPit/assets/123726577/04be4f91-d6e5-4dae-87d7-bd8717fc4d0e)

We have also startetd to add our template for the recipe upload and the login and registration. The registration is now working. The login and the registration page were designed using Axios in Node.js. Now the data is stored in a database. The template for uploading the recipe is still in progress.

That was our progress in this sprint, your CookPit Team.

**Comment by Stev3eM** [comment]
Hey CookPit Team,

sounds like your project keeps going forward and you already have reached a milestone when your API is finally connected and your database is working.

To the fact that all ingredients are stored in the form of a single string: Is it a direct copy of the AI output or is the recipe splitted into different parts within the json file and you need to extract several fields?

The function to avoid empty API calls is a good idea as every call will produce cost. To make it more effective you could implement further "no API rules" like: The recipe must contain a minimum number of letters to prevent only sending one letter, numbers or emojis.

Your start page looks great and I'm very excited of learning how to bake a Schokokuchen in just 5min!

Keep on going! 

**Your**   <img width="50" alt="image" src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/1a218c37-d3b0-4c88-9ae1-827b5a4eba42">   **Team**
